doc/sai-ptf/vlan_test_plan.md

```diff
@@ -0,0 +1,352 @@
+- Vlan functionalities
+- SAI APIs operations
+- Composit scenario
+
+# Test Environment
+## Testbed
+Those tests will be run on the testbed structure, the components are:
+* PTF - running in a server that can connect to the target DUT
+* SAI server - running on a dut
+## Test Configuration
+
+For the test configuration, please refer to the file 
+  - [VLAN_config](./config_data/vlan_config_t0.md)
+  - [FDB_config](./config_data/fdb_config_t0.md)
+  - [Route_config](./config_data/route_config_t0.md)
+  
+**Note. All the tests will be based on the configuration above, if any additional configuration is required, it will be specified in the Test case.**
+
+*p.s. Please refer the sample packet in [VLAN_config#Sample_Packet](./config_data/vlan_config_t0.md#sample-datapacket)*
+
+## Variations
+Cause the testbed might also encounter some issues like the host interface being down. 
+Before running the actual test there will need some sanity test to check the DUT status and select the active ports for testing.
+
+**All the ports in this test plan just to illustrate the test purpose, they are not exactly the same for the actual environment.**
+
+# Test Execution
+## Basic function test
+
+### Untagging and access
+
+#### Testing Objective
+This test verifies the VLAN function around untag and access ports.
+
+*p.s. This test will not check function with the native VLAN scenario (it will be in other tests). Please make sure the native VLAN will not impact the result.*
+
+
+
+With an untagged packet, on the access port, when ingress and egress happen, the behavior as below
+| Port mode | packet tag mode | Direction | Action                                   |
+| :---------|--------------- | :-------- | :--------------------------------------- |
+| Access|Untag| Ingress   | Accept the packet.       |
+|       |Untag | Egress    | Untag the packet. |
+| Access|Tag| Ingress   | Accept the packet.       |
+|       |Tag | Egress    | Untag the packet. |
+
+
+```
+Test example:
+Untagged:                                    
+  pkt(Untag:DMAC=MAC2)  -> Port1:Access:VLAN1000-> Port2:Access:VLAN1000 -> pkt(Untag:DMAC=MAC2)
+Tagged:
+  pkt(Tag:VLAN2000:DMAC=MAC10)  -> Port9:Access:VLAN2000-> Port10:Access:VLAN2000 -> pkt(Untag:DMAC=MAC10)
+```
+
+#### Test Data/Packet
+[Sample_Packet](./config_data/vlan_config_t0.md#sample-datapacket)
+- Input: Packet(Untag) and Packet(Tag)
+- Expected: Packet(Tag)
+
+  *p.s. please refer the sample packet in [VLAN_config#Sample_Packet](./config_data/vlan_config_t0.md#sample-datapacket)*
+
+#### Test Cases
+| Goal| Cases | Expect  |
+|-|-|-|
+|``Untag``:Acess -> Acess port.| Send VLAN100 ``Untagged`` packet on Port1 with mac2 as dest mac. |  ``Untag`` packet received on port2.|
+|``Tag``:Acess -> Acess port.| Send VLAN2000 ``Untagged`` packet on Port9 with mac10 mac as dest mac. |  ``Untag`` packet received on port10.|
+
+### Frame Filtering
+#### Testing Objective
+
+Drop packet when packet's dest mac is port SRC MAC in MAC table.
```

Comment:
Maybe change to 
Drop packet when the destination port from MAC table search is the port which packet come into the switch.